docs/proposals/buildstrategy-steps-resources.md

```diff
@@ -0,0 +1,197 @@
+steps resources is require, we need a more flexible way to deal with N number of steps under the same strategy (_Build/ClusterBuildStrategy_). Also, we still need to help users to know that independently of the abstraction of the strategies, there are options for speeding up the builds. This options could be presented in the form of different `flavours` of the same strategy(_cluster or namespaced scope_), that differ between each other in terms of container resources values.
+
+### Goals
+
+- Keep abstracting the `end-user` from the strategies content.
+
+- Allow `strategy admins` to set different flavours of the **same** strategies. This is the `strategy admin` responsability, and these flavours may vary depending on the Cloud provider. Recommendations
+of different flavours for each strategy could be made, based on the steps and the notion around, which of them consume higher numbers for cpu/memory.
+
+- Set `default` strategies flavours in this repository, so that all strategy containers have limits on the resources they consume, avoiding OOM and CPU throttling when running in Travis.
+
+- Modify the `Build` and the `BuildRun` API, to remove the `spec.resources` configuration path.
+
+- Ensure that the `resources` defined in each `_Build/ClusterBuildStrategy_` will actually be processed and passed on runtime to the generated `TaskRun` from Tekton.
+
+### Non-Goals
+
+- Defining the concrete numbers for the resources(_e.g. CPU/memory_) used per steps in the different Build/ClusterBuildStrategy strategies.
+
+## Proposal
+
+This proposal will consist of different steps:
+
+1. Remove the `spec.resources` from the `Build` and `BuildRun` API.
+
+2. Remove the logic inside the `generate_taskrun.go` and test classes, where we parse resources defined in the `Build/BuildRun` into the strategy containers. Ensure `resources` of the strategies are properly used.
+
+3. For all the current strategies(_BuildStrategy or ClusterBuildStrategy_) in this repository (_samples_), we should define default values for the container resources limits/request so that our `travis` will not be affected by strategy containers consuming all the CPU and memory (_unlimited_). Also, update all of our sample `Build` YAMLs to use this strategies.
+
+4. Properly document under the strategies docs, that all of the steps resources can be modified by the `strategy admin`. If a difference of values per strategy is needed, then multiple `ClusterBuildStrategy` will be required to exist on the cluster, with different names.
+
+5. Document different flavours for each `ClusterBuildStrategy` (_e.g. kaniko, buildpacks_). Each flavour should carefully illustrate how it differentiate with the default `ClusterBuildStrategy` values, and which advantage one can have when choosing this flavour under their `Build` resource. Advantages should be for now `faster builds`.
```

Comment:
As advantages, do we just mention `faster builds` or may also `larger builds`. Rational: a default build strategy might limit the memory to an amount that is too low for a build of a project that needs a lot memory (be it the compilation of a complex angular app or of a java app with a lot of binary dependencies; those come to my mind as memory-hungry).

Reply:
If a larger build is too slow when using the default flavor, moving to a flavor that provides more resources will end up in a faster build, therefore `faster build` is the main advantage.

Reply:
Well the large build might not be too slow but fail due to oom.